modules/andes-core/broker/src/main/java/org/wso2/andes/server/AMQChannel.java

```diff
@@ -299,6 +299,10 @@ public ConfigStore getConfigStore()
      */
     public void resendRecoveredMessages(DisruptorEventCallback recoverOKCallback) {
         try {
+
+            //messages should be removed from the unacknowledged map upfront
+            _unacknowledgedMessageMap.clear();
```

Comment:
I thinks it is better to call `_unacknowledgedMessageMap.cancelAllMessages()` which also reset the `_unackedSize`